### ISSUE TITLE
Compilation errors: void functions should not return values

### ISSUE THREAD

**Issue by moble** [numpy/numpy-dtypes#7]
On compilation, I get various errors of the following type:

```
npytypes/rational/rational.c:1068:9: error: void function 'initrational' should not return a value [-Wreturn-type]
        return NULL;
        ^      ~~~~
```

and

```
npytypes/quaternion/numpy_quaternion.c:529:9: error: void function 'initnumpy_quaternion' should not return a value [-Wreturn-type]
        return NULL;
        ^      ~~~~
```

If I simply change each of those lines to `return;`, compilation succeeds, and the modules seem to (mostly) work.

`python --version` = Python 2.7.7 
`numpy.__version__` = 1.8.1
`gcc --version` = Apple LLVM version 5.1 (clang-503.0.40) (based on LLVM 3.4svn)

**Comment by weiyw16** [comment]
have you fixed it ?

**Comment by moble** [comment]
It looks like this repo is dead (and was even when I opened this issue).  But I have created [a more complete `quaternion` package](https://github.com/moble/quaternion/) that gets tested routinely against python 2.7 and more recent 3.x versions.  The solution I use there is to detect the python version, and change the return statement as needed.  This can be seen starting at [this line](https://github.com/moble/quaternion/blob/master/numpy_quaternion.c#L1409).

If you want quaternions, you'd be better off just using my package.  If you want rationals, you might want to use my code as a starting point for a more up-to-date template.
